src/test/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/UploadDocumentsHandlerTest.java

```diff
@@ -0,0 +1,360 @@
+
+        assertNotNull(callbackResponse);
+        assertEquals(bailCase, callbackResponse.getData());
+
+        verify(bailCase, times(1)).read(UPLOAD_DOCUMENTS);
+
+        verify(documentReceiver, times(1)).tryReceive(homeOfficeDocumentBeingUploaded1,
+                                                      DocumentTag.UPLOAD_DOCUMENT, suppliedByHomeOffice);
+        verify(documentReceiver, times(1)).tryReceive(homeOfficeDocumentBeingUploaded2,
+                                                      DocumentTag.UPLOAD_DOCUMENT, suppliedByHomeOffice);
+
+        verify(documentsAppender, times(1)).append(existingHomeOfficeDocuments,
+                                                   homeOfficeDocumentsWithMetadataList);
+
+        verify(bailCase, times(1)).write(HOME_OFFICE_DOCUMENTS_WITH_METADATA,
+                                         allHomeOfficeDocuments);
+
+        verify(bailCase, times(1)).clear(UPLOAD_DOCUMENTS);
+        verify(bailCase, times(1)).clear(UPLOAD_DOCUMENTS_SUPPLIED_BY);
+    }
+
+    @Test
+    void should_not_change_uploaded_documents_if_not_present() {
+
+        when(bailCase.read(UPLOAD_DOCUMENTS)).thenReturn(Optional.empty());
+        when(callback.getCaseDetails().getCaseData()).thenReturn(bailCase);
+
+        assertDoesNotThrow(() -> uploadDocumentsDocumentHandler
+            .handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback));
+        PreSubmitCallbackResponse<BailCase> response = uploadDocumentsDocumentHandler
+            .handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
+
+        assertEquals(bailCase, response.getData());
+    }
+
+    @Test
+    void handling_should_throw_if_cannot_actually_handle() {
+
+        assertThatThrownBy(
+            () -> uploadDocumentsDocumentHandler.handle(PreSubmitCallbackStage.ABOUT_TO_START, callback))
+            .hasMessage("Cannot handle callback")
+            .isExactlyInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    void should_not_allow_null_arguments() {
+
+        assertThatThrownBy(() -> uploadDocumentsDocumentHandler.canHandle(null, callback))
+            .hasMessage("callbackStage must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(
+            () -> uploadDocumentsDocumentHandler.canHandle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, null))
+            .hasMessage("callback must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(() -> uploadDocumentsDocumentHandler.handle(null, callback))
+            .hasMessage("callbackStage must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(() -> uploadDocumentsDocumentHandler.handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, null))
+            .hasMessage("callback must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+    }
```

Comment:
Can  you please add a test for invalid suppliedBy value, to test the exception.

Reply:
Added